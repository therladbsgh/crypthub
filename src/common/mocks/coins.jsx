export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
