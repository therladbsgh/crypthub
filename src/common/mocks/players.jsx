import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
