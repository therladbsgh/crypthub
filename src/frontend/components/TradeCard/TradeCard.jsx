import React, { Component } from 'react';
import { Card, Header, Table } from 'semantic-ui-react';
import { TradeModal, Searchbar } from 'components';
import { SharedStyle as styles } from 'styles';

export default class TradeCard extends Component {
	constructor(props) {
		super(props);
		this.state = {
			results: ['RESET']
		};

		this.handleResults = this.handleResults.bind(this);
	}

	handleResults(results) {
		this.setState({ results });
	}

	render() {
		const { game, playerId, coins } = this.props;
		const { id, limitOrders, shortSelling, stopOrders } = game;
		const { results } = this.state;

		const orderTypes = [
			{
				text: 'Market Order',
				value: 'market'
			},
			...limitOrders ? [{
				text: 'Limit Order',
				value: 'limit'
			}] : [],
			...shortSelling ? [{
				text: 'Short Sell',
				value: 'short'
			}] : [],
			...stopOrders ? [{
				text: 'Stop Order',
				value: 'stop'
			}] : [],
		];

		return (
			<Card>
				<Card.Content className={styles.center}>
					<Header as='h2'>Search/Trade a Coin</Header>
					<Searchbar input={{ fluid: true }} placeholder='Coin name or symbol' source={coins} field='name' searchFields={['name', 'symbol']} handleResults={this.handleResults} open={false} />
					{!_.isEmpty(results) ?
						<Table striped>
							<Table.Body>
								{_.map(coins, (c, index) => 
									(_.some(results, _.mapKeys(_.mapValues(c, v => String(v)), (v, k) => k.toLowerCase())) || results[0] === 'RESET') &&
										<Table.Row key={index}>
											<Table.Cell>{c.symbol}</Table.Cell>
											<Table.Cell textAlign='right'><TradeModal coin={c} orderTypes={orderTypes} gameId={id} playerId={playerId} /></Table.Cell>
										</Table.Row>
								)}
							</Table.Body>
						</Table>
						:
						<Header as='h3'>No coins found</Header>
					}
				</Card.Content>
		  	</Card>
		);
  	}
}
