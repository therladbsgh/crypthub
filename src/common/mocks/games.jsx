import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
