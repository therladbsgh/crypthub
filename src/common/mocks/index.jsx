import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
