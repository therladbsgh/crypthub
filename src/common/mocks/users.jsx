import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
