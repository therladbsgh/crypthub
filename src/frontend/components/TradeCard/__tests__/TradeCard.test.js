import React from 'react';
import TradeCard from '../TradeCard';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<TradeCard game={GameMocks.game1} />).toJSON();
  	expect(tree).toMatchSnapshot();
});