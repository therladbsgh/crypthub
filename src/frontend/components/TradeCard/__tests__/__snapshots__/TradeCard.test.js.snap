// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <h2
      className="ui header"
    >
      Search/Trade a Coin
    </h2>
    <div
      className="ui fluid search"
      field="name"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      source={undefined}
    >
      <div
        className="ui fluid icon input"
      >
        <input
          autoComplete="off"
          className="prompt"
          disabled={undefined}
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Coin name or symbol"
          tabIndex="0"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          className="search icon"
        />
      </div>
      <div
        className="results transition"
      >
        <div
          className="message empty"
        >
          <div
            className="header"
          >
            No results found.
          </div>
        </div>
      </div>
    </div>
    <table
      className="ui striped table"
    >
      <tbody
        className=""
      />
    </table>
  </div>
</div>
`;
